### PR TITLE
Fix msg.value validation related revert messages in KyberProxy. remov…

### PR DESCRIPTION
**Pull request by ilanDoron**
…e over lapping message in kyberNetwork, update un reachable revert messages in kyberNetwork to assert

### CODE REVIEW

File: contracts/sol6/KyberNetworkProxy.sol [hunk 1]
```diff
@@ -375,7 +375,11 @@ contract KyberNetworkProxy is
         uint256 srcAmount,
         address destAddress
     ) internal returns (UserBalance memory balanceBefore) {
-        require(src == ETH_TOKEN_ADDRESS || msg.value == 0, "msg.value should be 0");
+        if (src == ETH_TOKEN_ADDRESS) {
+            require(msg.value == srcAmount, "Sent eth != srcAmount");
```
Comment (Anyhowclick) [inline contracts/sol6/KyberNetworkProxy.sol:379]: I think u mentioned etherscan doesn't show `!=` properly? Or was it `>` / `<`?

File: contracts/sol6/KyberNetwork.sol [hunk 2]
```diff
@@ -814,9 +814,8 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
         tradeData.networkFeeWei =
             (((tradeData.tradeWei * tradeData.networkFeeBps) / BPS) * tradeData.feeAccountedBps) /
             BPS;
-        require(
-            tradeData.tradeWei >= (tradeData.networkFeeWei + tradeData.platformFeeWei),
-            "fees exceed trade"
+        assert(
```
Comment (Anyhowclick) [inline contracts/sol6/KyberNetwork.sol:817]: can this be 1 line? or linter complains?

File: contracts/sol6/KyberNetworkProxy.sol (hunk 1 above)
Comment (ilanDoron) [reply contracts/sol6/KyberNetworkProxy.sol]: fixed

File: contracts/sol6/KyberNetwork.sol [hunk 3]
```diff
@@ -1026,15 +1025,15 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
         require(input.src != input.dest, "src = dest");
 
         if (input.src == ETH_TOKEN_ADDRESS) {
-            require(msg.value == input.srcAmount, "bad eth qty");
+            require(msg.value == input.srcAmount); // kyberNetworkProxy issues message here
         } else {
-            require(msg.value == 0, "eth not 0");
+            require(msg.value == 0); // kyberNetworkProxy issues message here
```
[the remark this answers is not in the document]
Comment (ilanDoron) [reply contracts/sol6/KyberNetwork.sol]: fixed

File: contracts/sol6/KyberNetwork.sol [hunk 4]
```diff
@@ -1026,15 +1025,15 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
         require(input.src != input.dest, "src = dest");
 
         if (input.src == ETH_TOKEN_ADDRESS) {
-            require(msg.value == input.srcAmount, "bad eth qty");
+            require(msg.value == input.srcAmount); // kyberNetworkProxy issues message here
```
[the remark this answers is not in the document]
Comment (ilanDoron) [reply contracts/sol6/KyberNetwork.sol]: fixed

File: contracts/sol6/KyberNetwork.sol [hunk 5]
```diff
@@ -1026,15 +1025,15 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
         require(input.src != input.dest, "src = dest");
 
         if (input.src == ETH_TOKEN_ADDRESS) {
-            require(msg.value == input.srcAmount, "bad eth qty");
+            require(msg.value == input.srcAmount); // kyberProxy issues message here
```
Comment (manhlx3006) [inline contracts/sol6/KyberNetwork.sol:1028]: The new proxy does, but the current proxy doesn't issue revert message for this case

Comment (ilanDoron) [reply contracts/sol6/KyberNetwork.sol:1028]: Yes

Got it

File: contracts/sol6/KyberNetworkProxy.sol [hunk 6]
```diff
@@ -375,7 +375,11 @@ contract KyberNetworkProxy is
         uint256 srcAmount,
         address destAddress
     ) internal returns (UserBalance memory balanceBefore) {
-        require(src == ETH_TOKEN_ADDRESS || msg.value == 0, "msg.value should be 0");
+        if (src == ETH_TOKEN_ADDRESS) {
+            require(msg.value == srcAmount, "Sent eth not equal to srcAmount");
```
Comment (manhlx3006) [inline contracts/sol6/KyberNetworkProxy.sol:379]: look inconsistent revert messages, we don't use uppercase for first letter :/

Comment (ilanDoron) [reply contracts/sol6/KyberNetworkProxy.sol]: right
thanks

File: contracts/sol6/KyberNetwork.sol (hunk 2 above)
Comment (ilanDoron) [reply contracts/sol6/KyberNetwork.sol]: good idea
done

**Pull request merged**